pepper-apis/src/main/java/org/broadinstitute/ddp/event/pubsubtask/api/PubSubTaskMessageParser.java

```diff
@@ -0,0 +1,113 @@
+package org.broadinstitute.ddp.event.pubsubtask.api;
+
+import static java.lang.String.format;
+import static org.broadinstitute.ddp.event.pubsubtask.api.PubSubTask.ATTR_PARTICIPANT_GUID;
+import static org.broadinstitute.ddp.event.pubsubtask.api.PubSubTask.ATTR_STUDY_GUID;
+import static org.broadinstitute.ddp.event.pubsubtask.api.PubSubTask.ATTR_TASK_TYPE;
+import static org.broadinstitute.ddp.event.pubsubtask.api.PubSubTask.ATTR_USER_ID;
+import static org.broadinstitute.ddp.event.pubsubtask.api.PubSubTaskLogUtil.errorMsg;
+import static org.broadinstitute.ddp.event.pubsubtask.api.PubSubTaskLogUtil.infoMsg;
+import static org.slf4j.LoggerFactory.getLogger;
+
+import java.util.Map;
+
+
+import com.google.cloud.pubsub.v1.AckReplyConsumer;
+import com.google.gson.Gson;
+import com.google.pubsub.v1.ProjectSubscriptionName;
+import com.google.pubsub.v1.PubsubMessage;
+import org.broadinstitute.ddp.util.GsonUtil;
+import org.slf4j.Logger;
+
+/**
+ * Parses {@link PubSubTask} from subscription
+ * defined by config param "pubsub.pubSubTasksSubscription".
+ * The result {@link PubSubTaskMessageParseResult} contains
+ * created {@link PubSubTask} with parsed message.
+ * If error occured during parsing then {@link PubSubTaskMessageParseResult}
+ * contains non-null error message (to be sent to result topic).
+ */
+public class PubSubTaskMessageParser {
+
+    private static final Logger LOG = getLogger(PubSubTaskMessageParser.class);
+
+    private final ProjectSubscriptionName projectSubscriptionName;
+    private final PubSubTaskProcessorFactory pubSubTaskProcessorFactory;
+
+    private final Gson gson = GsonUtil.standardGson();
+
+    public PubSubTaskMessageParser(ProjectSubscriptionName projectSubscriptionName,
+                                   PubSubTaskProcessorFactory pubSubTaskProcessorFactory) {
+        this.projectSubscriptionName = projectSubscriptionName;
+        this.pubSubTaskProcessorFactory = pubSubTaskProcessorFactory;
+    }
+
+    public PubSubTaskMessageParseResult parseMessage(PubsubMessage message, AckReplyConsumer consumer) {
+        String messageId = message.getMessageId();
+        String taskType = message.getAttributesOrDefault(ATTR_TASK_TYPE, null);
+        String participantGuid = message.getAttributesOrDefault(ATTR_PARTICIPANT_GUID, null);
+        String userId = message.getAttributesOrDefault(ATTR_USER_ID, null);
+        String studyGuid = message.getAttributesOrDefault(ATTR_STUDY_GUID, null);
```

Comment:
I thought all of these fields (other than `messageId`) were going to be fields in the payload?  My concern here is that it may be difficult to re-run an operation using these as message attributes when compared to deserializing the data payload.

Reply:
@zyme you mean by re-run - to call this task not from DSM but for example from command line?

Reply:
@zyme could we discuss this suggestion (to put all data in the payload) together with DSM team? I posted the message about it in Slack channel ddp-5235. Because they also needs to do these changes in case if it will be decided to do it.